### PR TITLE
[DUOS-2187][risk=no] Add form to edit datasets

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2187

### Summary

Add form to edit datasets

Screenshot 2023-06-16 at 14-00-55 Broad Data Use Oversight System

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/data_update/DatasetUpdate.js [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+
+    const formElement = event.target.form;
+    const submitData = new FormData(formElement);
+    const nihFileData = submitData.get('nihInstitutionalCertificationFile');
+    const consentGroups = [{ nihInstitutionalCertificationFile: nihFileData }];
+
+    const newDataset = {
+      name: dataset.name,
+      dacId: formData.dac.dacId,
+      properties: [
```
Comment (fboulnois) [inline src/components/data_update/DatasetUpdate.js:84]: Although the OpenAPI schema for DatasetUpdate says it requires a `datasetProperties` object, the `DatasetUpdate` object in Consent actually requires a `properties` object instead.

File: src/components/data_update/DatasetUpdate.js [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+        asProperty('Data Depositor', formData.properties.dataDepositor),
+        asProperty('Principal Investigator(PI)', formData.properties.principalInvestigator),
+      ]
+    };
+
+    const multiPartFormData = new FormData();
+    multiPartFormData.append('dataset', JSON.stringify(newDataset));
+    multiPartFormData.append('consentGroups', consentGroups);
+
+    DataSet.updateDatasetV3(dataset.dataSetId, multiPartFormData).catch(() => {
```
Comment (fboulnois) [inline src/components/data_update/DatasetUpdate.js]: Additionally, although the response returns 200, the underlying dataset does not actually get updated. I think there's probably an issue with the `updateDataset` endpoint.

**Review by kaitlynoflaherty — APPROVED** [review]
The form looks great! When testing locally, everything works as expected besides the submit button. It seems like some required fields have been excluded from the endpoint, so I agree that the API should be updated.

File: src/components/data_update/DatasetUpdate.js (hunk 1 above)
Comment (rushtong) [reply src/components/data_update/DatasetUpdate.js:84]: Nice catch, we should file a bug fix for that.

File: src/components/data_update/DatasetUpdate.js (hunk 2 above)
Comment (rushtong) [reply src/components/data_update/DatasetUpdate.js]: I'm seeing some updates when using the submit button. For example, the dataset name updates, but the species doesn't seem to update. I think the properties are not saving correctly. What I think is missing is some UI indication of success.

**Review by rushtong — COMMENTED** [review]
The form looks good, but I think we need an indicator of success somewhere on the page. We use the noty library now for that kind of thing.

The back end does seem to not be saving the properties correctly. This did work in testing so we need to sort out what's happening there.

File: src/components/data_update/DatasetUpdate.js [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+    event.preventDefault();
+    event.stopPropagation();
+
+    const formElement = event.target.form;
+    const submitData = new FormData(formElement);
+    const nihFileData = submitData.get('nihInstitutionalCertificationFile');
+    const consentGroups = [{ nihInstitutionalCertificationFile: nihFileData }];
+
+    const newDataset = {
+      name: dataset.name,
```
Comment (rushtong) [inline src/components/data_update/DatasetUpdate.js]: I think this should align with the same value in `formData.properties.datasetName`. As it stands now, there are two different names being sent over in the payload. This is a bit of a holdover from previous technical debt where the name was not a top level field, but just a prop.

Comment (fboulnois) [reply src/components/data_update/DatasetUpdate.js]: Thanks, good to know, I wasn't sure which of these I should use.

File: src/components/data_update/DatasetUpdate.js (hunk 1 above)
Comment (fboulnois) [reply src/components/data_update/DatasetUpdate.js:84]: https://broadworkbench.atlassian.net/browse/DUOS-2579

File: src/components/data_update/DatasetUpdate.js (hunk 2 above)
Comment (fboulnois) [reply src/components/data_update/DatasetUpdate.js]: Just like the submission form, I added a success notification and a redirect to the catalog page.

**Review by rushtong — APPROVED** [review]
👍🏽

**Pull request merged**